app/code/core/Mage/CatalogInventory/Model/Observer.php

```diff
@@ -273,8 +273,8 @@ protected function _removeErrorsFromQuoteAndItem($item, $code)
         $canRemoveErrorFromQuote = true;
 
         /** @var Mage_Sales_Model_Quote_Item $quoteItem */
-        foreach ($quoteItems as $quoteItem) {
-            if ($quoteItem->getItemId() == $item->getItemId()) {
+        foreach ($quoteItems as $quoteItemId => $quoteItem) {
```

Comment:
where is the ID coming from in the Collection? As far as I see its build in most of the cases without the itemId as key.

Reply:
would an alternate be to actually code getItemId() instead of relying on the __call() method?

Reply:
It's seems that majority of collection have the "id" as key?

From [lib/Varien/Data/Collection.php](https://github.com/OpenMage/magento-lts/blob/1.9.4.x/lib/Varien/Data/Collection.php), there are some references for id used as key. But perhaps I'm wrong.

From the method `load` of a collection, there is something about the id:
https://github.com/OpenMage/magento-lts/blob/fc98af9ed61e6fc225ac0e864a5fdf564aac1a36/lib/Varien/Data/Collection/Db.php#L583-L592

https://github.com/OpenMage/magento-lts/blob/fc98af9ed61e6fc225ac0e864a5fdf564aac1a36/lib/Varien/Data/Collection.php#L359-L365

Reply:
The item id is indeed the key for all collections.